### DISCUSSION TITLE
组件库如何适配 Vue 2.7

### DISCUSSION THREAD

**Discussion by xiaosansiji** [Tencent/tdesign#278] [General]
## 现状

- [tdesign-vue](https://github.com/Tencent/tdesign-vue) 可在 Vue 2.6 项目中运行
- [tdesign-vue-next](https://github.com/Tencent/tdesign-vue-next) 可在 Vue 3.x 项目中运行

对于 Vue 社区推出的，旨在引导 Vue 2 用户平滑升级到 Vue 3 的 Vue 2.7 版本，TDesign 目前暂无组件库能够正常运行。

## 期望

能够用现有 tdesign-vue 或 tdesign-vue-next 组件适配到 Vue 2.7.x 环境。

### tdesign-vue 适配 Vue 2.7

tdesign-vue 中额外引入了 `@vue/composition-api` https://github.com/Tencent/tdesign-vue/blob/develop/package.json#L91
已经有部分组件用 `@vue/composition-api` 提供的语法重写了实现，与对应的 tdesign-vue-next 仓库组件共享逻辑实现，降低同时维护的成本。

1. 全局检查：引入组件使用时会默认检查并注册 `@vue/composition-api` https://github.com/Tencent/tdesign-vue/blob/develop/src/index.ts#L5
2. 组件内部实现：直接从 `@vue/composition-api` 引入 https://github.com/Tencent/tdesign-vue/blob/develop/src/menu/menu.tsx#L1

参照 [Vue 2.7.0 changelog](https://github.com/vuejs/vue/blob/main/CHANGELOG.md#270-2022-07-01)

Vue 2.7.x 中已经默认包含了 Composition API，不需要额外引入 `@vue/composition-api`，上述两处实现需要做调整：
1. 全局检查或许可以增加当前 Vue 版本判断，Vue 2.7.x 版本下不再做 `Vue.use(VueCompositionAPI)` 处理
2. 组件内部实现时引入包名从 `import { x } from '@vue/composition-api'` 改到 `import { x } from 'vue'`

2 不确定是否在用户引入组件库时通过增加其他额外插件或者配置的方式来修改更好。

### tdesign-vue-next 适配 Vue 2.7

2.7 版本下缺少如下 `import { xx } from 'vue'` 的内容：
- createVNode
- isVNode
- mergeProps
- withDirectives
- resolveDirective
- vShow
- Transition
- Fragment
- createTextVNode

## more

如果有其他验证存在的适配问题，或者可行的方案，可以补充在评论中，有切实可行提案后可以启动相关组件库适配改造。

**Comment by littleCareless** [comment]
请问还在推进吗？
